File: backend/src/modules/user/user.controller.ts
import { Body, Controller, Get, Headers, Post, UseGuards } from '@nestjs/common';
import { HeaderAuthGuard, Validation } from '../../common/auth/header-auth.guard';
import { CurrentUser } from '../../common/auth/current-user.decorator';
import { UserService, UserAdminDto } from './user.service';
import { LoginDto } from './dto/login.dto';

// Route base is /User (no api/ prefix — matches the documented contract).
@Controller('User')
export class UserController {
  constructor(private readonly service: UserService) {}

  @Post('Login')
  login(@Body() dto: LoginDto) {
    return this.service.login(dto);
  }

  @Post('Logout')
  logout(@Headers('usertoken') token?: string) {
    return this.service.logout(token);
  }

  @Get('GetUserData')
  @UseGuards(HeaderAuthGuard)
  getUserData(@CurrentUser() v: Validation) {
    return this.service.getUserData(v.userId);
  }

  /** GET /User/GetAllUsers — login users (with roles) for supervisor pickers. */
  @Get('GetAllUsers')
  @UseGuards(HeaderAuthGuard)
  getAllUsers() {
    return this.service.getAllUsers();
  }

  /** GET /User/GetUserList — paginated users for the Users admin screen. */
  @Get('GetUserList')
  @UseGuards(HeaderAuthGuard)
  getUserList(
    @Headers('pageno') pageNo?: string,
    @Headers('noofitems') noOfItems?: string,
    @Headers('roleid') roleId?: string,
  ) {
    return this.service.getUserList(Number(pageNo) || 1, Number(noOfItems) || 20, Number(roleId) || undefined);
  }

  /** POST /User/AddUser — create a login user with a role. */
  @Post('AddUser')
  @UseGuards(HeaderAuthGuard)
  addUser(@Body() body: UserAdminDto) {
    return this.service.addUser(body);
  }

  /** POST /User/UpdateUser — edit a login user's profile, role and status. */
  @Post('UpdateUser')
  @UseGuards(HeaderAuthGuard)
  updateUser(@Body() body: UserAdminDto) {
    return this.service.updateUser(body);
  }

  /** POST /User/DeleteUser — deactivate a login user (Id header). */
  @Post('DeleteUser')
  @UseGuards(HeaderAuthGuard)
  deleteUser(@Headers('id') id?: string) {
    return this.service.deleteUser(Number(id));
  }
}

File: backend-dotnet/GarasTransport.Api/Modules/User/UserController.cs
using GarasTransport.Api.Common;
using Microsoft.AspNetCore.Mvc;

namespace GarasTransport.Api.Modules.User;

// Route base is /User (no api/ prefix — matches the documented contract).
[Route("User")]
public class UserController : ApiControllerBase
{
    private readonly UserService _service;

    public UserController(UserService service) => _service = service;

    [HttpPost("Login")]
    public Task<object> Login([FromBody] LoginDto dto) => _service.Login(dto);

    [HttpPost("Logout")]
    public Task<object> Logout() => _service.Logout(Header("UserToken"));

    [HttpGet("GetUserData")]
    [HeaderAuth]
    public Task<object> GetUserData() => _service.GetUserData(CurrentUser.UserId);

    /// <summary>GET /User/GetAllUsers — login users (with roles) for supervisor pickers.</summary>
    [HttpGet("GetAllUsers")]
    [HeaderAuth]
    public Task<object> GetAllUsers() => _service.GetAllUsers();

    /// <summary>GET /User/GetUserList — paginated users for the Users admin screen.</summary>
    [HttpGet("GetUserList")]
    [HeaderAuth]
    public Task<object> GetUserList() =>
        _service.GetUserList(HeaderInt("PageNo", 1), HeaderInt("NoOfItems", 20), HeaderIntOrNull("RoleId"));

    /// <summary>POST /User/AddUser — create a login user with a role.</summary>
    [HttpPost("AddUser")]
    [HeaderAuth]
    public Task<object> AddUser([FromBody] UserAdminDto body) => _service.AddUser(body);

    /// <summary>POST /User/UpdateUser — edit a login user's profile, role and status.</summary>
    [HttpPost("UpdateUser")]
    [HeaderAuth]
    public Task<object> UpdateUser([FromBody] UserAdminDto body) => _service.UpdateUser(body);

    /// <summary>POST /User/DeleteUser — deactivate a login user (Id header).</summary>
    [HttpPost("DeleteUser")]
    [HeaderAuth]
    public Task<object> DeleteUser() => _service.DeleteUser(HeaderInt("Id", 0));
}
